core/src/com/unciv/models/ruleset/GameSpeed.kt

```diff
@@ -0,0 +1,98 @@
+    override fun getCivilopediaTextLines(ruleset: Ruleset) = sequence {
+        yield(FormattedLine("General speed modifier: [${modifier * 100}]%${Fonts.turn}"))
+        yield(FormattedLine("Production cost modifier: [${productionCostModifier * 100}]%${Fonts.production}"))
+        yield(FormattedLine("Gold cost modifier: [${goldCostModifier * 100}]%${Fonts.gold}"))
+        yield(FormattedLine("Science cost modifier: [${scienceCostModifier * 100}]%${Fonts.science}"))
+        yield(FormattedLine("Culture cost modifier: [${cultureCostModifier * 100}]%${Fonts.culture}"))
+        yield(FormattedLine("Faith cost modifier: [${faithCostModifier * 100}]%${Fonts.faith}"))
+        yield(FormattedLine("Improvement build length modifier: [${improvementBuildLengthModifier * 100}]%${Fonts.turn}"))
+        yield(FormattedLine("Diplomatic deal duration: [$dealDuration] turns${Fonts.turn}"))
+        yield(FormattedLine("Gold gift influence gain modifier: [${goldGiftModifier * 100}]%${Fonts.gold}"))
+        yield(FormattedLine("Barbarian spawn modifier: [${barbarianModifier * 100}]%${Fonts.strength}"))
+        yield(FormattedLine("Golden age length modifier: [${goldenAgeLengthModifier * 100}]%${Fonts.happiness}"))
+        yield(FormattedLine("Adjacent city religious pressure: [$religiousPressureAdjacentCity]${Fonts.faith}"))
+        yield(FormattedLine("Peace deal duration: [$peaceDealDuration] turns${Fonts.turn}"))
+    }.toList()
+
+    fun numTotalTurns(): Int = yearsToTurnObject.last().toTurn
+
+    private fun initYearsToTurn(yearsToTurn: ArrayList<HashMap<String, Float>>): ArrayList<YearsPerTurn> {
+        val yptList = ArrayList<YearsPerTurn>()
+
+        for (incrementMap: HashMap<String, Float> in yearsToTurn) {
+            if (incrementMap["turnsPerIncrement"] == null || incrementMap["yearsPerTurn"] == null) continue
+            val runningSum = incrementMap["turnsPerIncrement"]!!.toInt() + if (yptList.size > 0) yptList.last().toTurn else 0
+            yptList.add(YearsPerTurn(incrementMap["yearsPerTurn"]!!, runningSum))
+        }
+        return yptList
+    }
+
+    fun initDefaultPercents() {
+        // checks for a non-negative `modifier` value when loading a mod ruleset
+        // this way the user should see an error popup
+        if (goldCostModifier < 0f) goldCostModifier = modifier
+        if (productionCostModifier < 0f) productionCostModifier = modifier
+        if (scienceCostModifier < 0f) scienceCostModifier = modifier
+        if (cultureCostModifier < 0f) cultureCostModifier = modifier
+        if (faithCostModifier < 0f) faithCostModifier = modifier
+        if (improvementBuildLengthModifier < 0f) improvementBuildLengthModifier = modifier
+        if (goldGiftModifier < 0f) goldGiftModifier = modifier
+        if (barbarianModifier < 0f) barbarianModifier = modifier
+        if (goldenAgeLengthModifier < 0f) goldenAgeLengthModifier = modifier
+    }
```

Comment:
I think that error popup comment was a classic case of me starting to type something, stopping for a moment to think, then getting distracted and never finishing it since I don't even know what I was talking about (maybe the check that happens when a ruleset is loaded).